test/NuGet.Clients.Tests/NuGet.CommandLine.Test/PortReserver.cs

```diff
@@ -28,14 +28,16 @@ public class PortReserver : IDisposable
         // Since practically there is no perf issue or concern here, this keeps the code the simplest possible.
         private static HashSet<int> _appDomainOwnedPorts = new HashSet<int>();
 
-        public int PortNumber
-        {
-            get;
-            private set;
-        }
+        public int PortNumber { get; private set; }
+        public string BaseUri { get; }
 
-        public PortReserver(int basePort = 50231)
+        public PortReserver(string basePath = null, int basePort = 50231)
         {
+            if (!string.IsNullOrEmpty(basePath) && (!basePath.StartsWith("/") || basePath.EndsWith("/")))
+            {
+                throw new ArgumentException($"If provided, argument \"{nameof(basePath)}\" must start with and must not end with a slash (/)");
```

Comment:
Consider adding documentation to the constructor explaining correct usage of the parameters.